src/dune_pkg/local_package.ml

```diff
@@ -7,15 +8,91 @@ type t =
   ; loc : Loc.t
   }
 
-let to_opam_file { name; version; dependencies; loc = _ } =
-  OpamFile.OPAM.empty
-  |> OpamFile.OPAM.with_name (Package_name.to_opam_package_name name)
-  |> OpamFile.OPAM.with_version_opt
-       (Option.map version ~f:Package_version.to_opam_package_version)
-  |> OpamFile.OPAM.with_depends
-       (Package_dependency.list_to_opam_filtered_formula dependencies)
-;;
+module Dependency_hash = struct
+  module Hash = Sha256
```

Comment:
Another thing to be aware of is how to deal with hash migration. Maybe a scheme where the hash is prefixed by its algorithm like OPAMs `sha1=deadcafeb0ba` would make sense in places where the user is exposed to the hash?

Reply:
changed the format to `md5=<hash>`

Reply:
I'm sorry I'm a bit late to the party, but I think the hash algorithm should be controlled by the version of the lock directory set in `lock.dune`. The whole hash thing is a complete implementation detail of dune and there's no need to expose any details to the user.